File: scala/src/test/scala/com/shekhargulati/ninetynine_problems/scala/lists/P27Test.scala
package com.shekhargulati.ninetynine_problems.scala.lists

import org.scalatest.{FunSpec, Matchers}

class P27Test extends FunSpec with Matchers {

  describe("P27 Spec") {
    it("should group 9 person into 3 groups of size 2,3, and 4") {
      val group: List[(List[String], List[String], List[String])] = P27.group3(List("aldo", "beat", "carla", "david", "evi", "flip", "gary", "hugo", "ida"))
      group.foreach(println)
      println(group.size)
      group should have length 1260

    }
  }

}

File: java8/src/test/java/com/shekhargulati/ninetynine_problems/java8/lists/P27Test.java
package com.shekhargulati.ninetynine_problems.java8.lists;

import org.junit.Test;

import java.util.List;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;
import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
import static org.junit.Assert.assertThat;

public class P27Test {

    @Test
    public void shouldGroupIntoThreeGroupsOfSize_2_3_and_4() throws Exception {
        List<String> input = Stream.of("aldo", "beat", "carla", "david", "evi", "flip", "gary", "hugo", "ida").collect(toList());
        List<List<List<String>>> groups = P27.group3(input);
        assertThat(groups, hasSize(1260));
    }
}